apps/trading/components/vega-wallet-connect-button/vega-wallet-connect-button.tsx

```diff
@@ -12,14 +15,107 @@ import {
 } from '@vegaprotocol/ui-toolkit';
 import type { PubKey } from '@vegaprotocol/wallet';
 import { useVegaWallet, useVegaWalletDialogStore } from '@vegaprotocol/wallet';
-import { useEffect, useMemo, useState } from 'react';
-import CopyToClipboard from 'react-copy-to-clipboard';
+import { Networks, useEnvironment } from '@vegaprotocol/environment';
+
+const MobileWalletButton = ({ isConnected }: { isConnected?: boolean }) => {
+  const { pubKeys, disconnect } = useVegaWallet();
+  const openVegaWalletDialog = useVegaWalletDialogStore(
+    (store) => store.openVegaWalletDialog
+  );
+  const { VEGA_ENV } = useEnvironment();
+  const isYellow = VEGA_ENV === Networks.TESTNET;
+  const [drawerOpen, setDrawerOpen] = useState(false);
+  const mobileDisconnect = useCallback(() => {
+    setDrawerOpen(false);
+    disconnect();
+  }, [disconnect]);
+  const openDrawer = useCallback(() => {
+    if (isConnected) {
+      setDrawerOpen(!drawerOpen);
+    } else {
+      openVegaWalletDialog();
+      setDrawerOpen(false);
+    }
+  }, [drawerOpen, isConnected, openVegaWalletDialog]);
+
+  return (
+    <div className="md:hidden overflow-hidden flex">
+      <button
+        className={classNames(
+          'my-2 transition-all flex flex-col justify-around gap-3 p-2 relative z-20 h-[34px]',
+          { 'z-50': drawerOpen }
+        )}
+        onClick={openDrawer}
+        data-testid="connect-vega-wallet-mobile"
+      >
+        <svg
```

Comment:
it's because close button is exactly the same one which for opening. It has only changed internal content. Eventually we can change it for full animation, if will be such an idea in the future

Reply:
Hm I don't love this. The close button should always be in the top right (as per designs). I'd rather forgo the animation if needed.

Reply:
suggestion (non blocking): Additionally I think this would be simpler if there is a single drawer with different internal views (keys or pages) depending on what menu item is selected.

Reply:
so close has been moved to the right.
Not quite get it how it should look like, I mean a single drawer. We have a component similar to Dialog component, and its reusable. If we have two buttons for open two different drawers, it'd doesn't matter, I suppose, in how many components we put it. Or maybe the problem is both drawer stay mounted all the time in the DOM and stay useless on HD displays? If so maybe better would be to use dimension hook and condition mounting this on screen resolution?